addons/stock/migrations/11.0.1.1/post-migration.py

```diff
@@ -319,15 +320,18 @@ def recompute_stock_move_line_qty_different_uom(env):
     )
     lines = env['stock.move.line'].browse([x[0] for x in env.cr.fetchall()])
     for line in lines:
-        product_qty = line.product_uom_id._compute_quantity(
-            line.product_uom_qty, line.product_id.uom_id,
-            rounding_method='HALF-UP',
-        )
-        # Can't assign by ORM, so by SQL
-        env.cr.execute(
-            "UPDATE stock_move_line SET product_qty = %s WHERE id = %s",
-            (product_qty, line.id),
-        )
+        try:  # Don't fail on incompatible conversions
+            product_qty = line.product_uom_id._compute_quantity(
+                line.product_uom_qty, line.product_id.uom_id,
+                rounding_method='HALF-UP',
+            )
+            # Can't assign by ORM, so by SQL
```

Comment:
why can't assign by ORM?

Reply:
Because there's associated code to the `write` on stock.move.line that makes a lot of reserve/unreserve operations on quants that will mess up quant state.